### PR TITLE
[Merged by Bors] - ci: post a weekly report to zulip with some technical debt stats

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
Great idea! I'd like to suggest three further metrics, at least for the near future.
- number of documentation `nolints`s: almost all of them are genuine at this point
Computing them can be as simple as `cat scripts/nolints.json | grep -c "docBlame"`.
- number of large files (as flagged by the style linter): long-term, this metric may cease to be useful, as a few transient exceptions are presumably normal. In my opinion, we're not at this point yet.) Can be computed using `cat scripts/style-exceptions.txt | grep -c "ERR_NUM_LIN"` (currently 97); using `git grep` would also work.
- number of missing module docstrings: `cat scripts/style-exceptions.txt | grep -c "ERR_MOD"` (currently 31)
- number of files in `Mathlib/` with `autoImplicit true` : `git grep -c "autoImplicit true" | grep -c "Mathlib"`
(For all files, `git grep -c "autoImplicit true" | wc -l` will also work.)
Again, there may be a number of files where `autoImplicit true` is used sensibly throughout the entire file - but in my experience removing this from files, most files do not use this intentionally, as it's just another small piece of technical debt.)

**Comment by adomani** [comment]
There is also #13038 that has more statistics.

**Comment by jcommelin** [comment]
@grunweg would you like to add those to this PR? Or as a follow-up?

**Comment by semorrison** [comment]
Let's iterate fast. We'll merge this and add more later.

I'd love to see graphs, too. :-)

**Comment by semorrison** [comment]
bors merge

**Comment by grunweg** [comment]
I just pushed a commit adding the four metrics I proposed, not having seen the approval. Apologies if this cancelled the bors push.

**Comment by adomani** [comment]
Michael, I think that bors is struggling to merge a group of PRs with some conflicts among them, so even without your push this would have probably failed anyway.

File: scripts/technical-debt-counters.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    then fl="-i"  # just for porting notes we ignore the case in the regex
+    else fl="--"
+    fi
+    printf '| %s | %s |\n' "$(git grep "${fl}" "${regex}" | wc -l)" "${title}"
+  fi
+done
+
+printf '|%s | %s |\n' "$(cat scripts/nolints.json | grep -c 'docBlame')" "documentation nolint entries"
+printf '|%s | %s |\n' "$(cat scripts/style-exceptions.txt | grep -c 'ERR_MOD')" "missing module docstrings"
+printf '|%s | %s |\n' "$(cat scripts/style-exceptions.txt | grep -c 'ERR_NUM_LIN')" "large files"
```
Comment (adomani) [inline scripts/technical-debt-counters.sh:30]:
```suggestion
printf '|%s | %s |\n' "$(grep -c 'docBlame' scripts/nolints.json)" "documentation nolint entries"
printf '|%s | %s |\n' "$(grep -c 'ERR_MOD' scripts/style-exceptions.txt)" "missing module docstrings"
printf '|%s | %s |\n' "$(grep -c 'ERR_NUM_LIN' scripts/style-exceptions.txt)" "large files"
```

Comment (grunweg) [reply scripts/technical-debt-counters.sh]: That's even nicer, thanks!

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**